xpra/server/proxy/proxy_server.py

```diff
@@ -560,9 +560,15 @@ def unexpected_packet(packet):
     def start_new_session(self, username, _password, uid, gid, new_session_dict=None, displays=()):
         log("start_new_session%s", (username, "..", uid, gid, new_session_dict, displays))
         sns = typedict(new_session_dict or {})
-        mode = sns.get("mode", "start")
+        mode = "start"
+        display = None
+        for k,v in sns.items():
+            k = bytestostr(k)
+            if k in ("mode"):
+                mode = bytestostr(v)
+            if k in ("display"):
+                display = bytestostr(v)
```

Comment:
AFAICT, this does the exact same thing in a much more concise way: 50b6f4b605e593b1058fac7b180b049d1e5ab2fa

Reply:
Dropping this commit then, thanks! :+1: